mitodl/odl-video-service
# Dorska/video analytics frontend

#### What are the relevant tickets?
https://github.com/mitodl/odl-video-service/issues/402
https://github.com/mitodl/odl-video-service/issues/403

#### What's this PR do?
Adds frontend code for video analytics feature.
see: https://github.com/mitodl/odl-video-service/issues/402

#### How should this be manually tested?

##### 1. Test collections page w/ mock data
1. Go to a collection page that lists videos.
2. Open a developer console, and set `window.ovsMockAnalytics = 1`. This will cause the frontend to query the backend for 'mock' analytics data.
3. Click the triple-dot menu on any video card, and select `Analytics`. The analytics dialog should pop-up.
4. Ensure that the chart/table look reasonable, per 
https://github.com/mitodl/odl-video-service/issues/402#issuecomment-374749212

##### 2. Test VideoDetail page w/ mock data
1. Go to a video detail page (the page that has the video player)
2. Open a developer console, and set `window.ovsMockAnalytics = 1`. This will cause the frontend to query the backend for 'mock' analytics data.
3. Click the `Analytics` button. The analytics dialog should pop-up.
4. Ensure that the chart/table look reasonable, per 
https://github.com/mitodl/odl-video-service/issues/402#issuecomment-374749212

##### 3. Test w/ real data.
0. Setup a google analytics 'property'.
1. Add a dimension to your GA property. Call it 'dimesion1'. See: https://github.com/mitodl/odl-video-service/issues/401 . 
2. Setup a google service account and save its keyfile.
3. Add your service account to your GA property. (In GA, 'admin' => 'user management' under the property section).
4. add these settings to your `.env` file:
  - `GA_CAMERA_DIMENSION=dimension1` (from dimension above)
  - `GA_TRACKING_ID` (e.g. from GA account, `UA-115967001-1`)
  - `GA_VIEW_ID` (e.g. `171841111`) (from GA view)
  - `GA_KEYFILE_JSON` (json string for the service account keyfile)
5. Restart your local web server.
6. Go to Google Analytics, and verify that events were logged for your video watching.
7. Bring up an analytics dialog for your video. It should now display the real data, queried from Google Analytics.

#### Where should the reviewer start?
The meat of this feature is in:
`static/js/components/dialogs/AnalyticsDialog`
`static/js/components/analytics`
`static/js/reducers/videoAnalytics`
`static/js/lib/api.js`
